llvm/lib/Target/EVM/EVMConstantSpiller.cpp

```diff
@@ -0,0 +1,100 @@
+//===----- EVMConstantSpiller.cpp - Spill constants to memory --*- C++ -*--===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file identifies IMM_RELOAD instructions representing spilled constants
+// throughout the module. It spills constants at the start of the entry function
+// and replaces IMM_RELOAD with the corresponding reload instructions.
+//
+//===----------------------------------------------------------------------===//
+
+#include "EVMConstantSpiller.h"
+#include "EVMInstrInfo.h"
+#include "EVMSubtarget.h"
+#include "MCTargetDesc/EVMMCTargetDesc.h"
+#include "TargetInfo/EVMTargetInfo.h"
+#include "llvm/CodeGen/MachineFrameInfo.h"
+#include "llvm/CodeGen/MachineModuleInfo.h"
+#include "llvm/CodeGen/Passes.h"
+#include "llvm/CodeGen/TargetSubtargetInfo.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Support/Debug.h"
+
+using namespace llvm;
+
+#define DEBUG_TYPE "evm-spill-constants"
+
+constexpr uint64_t SpillSlotSize = 32;
+
+static MachineInstr *emitPush(MachineBasicBlock &MBB,
+                              MachineBasicBlock::iterator InsertBefore,
+                              const EVMInstrInfo *TII, const APInt &Imm,
+                              LLVMContext &Ctx, const DebugLoc &DL) {
+  unsigned Opc = EVM::getPUSHOpcode(Imm);
```

Comment:
Maintainability: Similar logic already exists in `EVMConstantUnfolding.cpp` and probably other `EVM::getPUSHOpcode` call sites. Can we unify it?